Modules/MUON/Common/src/TracksPostProcessing.cxx

```diff
@@ -0,0 +1,221 @@
+}
+
+template <class HIST>
+void MatchingEfficiencyPlotter<HIST>::update(repository::DatabaseInterface& qcdb, Trigger t, std::shared_ptr<o2::quality_control::core::ObjectsManager> objectsManager)
+{
+  std::string pathMCH = mPlotPath[0];
+  auto moMCH = getMO(qcdb, pathMCH, mPlotName[0], t, mTimestamp[0]);
+  std::string pathMatched = mPlotPath[1];
+  auto moMatched = getMO(qcdb, pathMatched, mPlotName[1], t, mTimestamp[1]);
+  // check if both MOs are existing and updated
+  if (!moMCH.second || !moMatched.second) {
+    return;
+  }
+
+  auto hMCH = getPlotFromMO<HIST>(moMCH.first);
+  auto hMatched = getPlotFromMO<HIST>(moMatched.first);
+  // check if both plots are valid
+  if (!hMCH || !hMatched) {
+    return;
+  }
+
+  TString name = mName.c_str();
+  auto title = TString::Format("%s - matching eff.", hMatched->GetTitle());
+
+  if (!mHistMatchingEff) {
+    HIST* h = dynamic_cast<HIST*>(hMatched->Clone());
+    if (!h) {
+      return;
+    }
+    h->Rebin(mRebin);
+    mHistMatchingEff.reset(h);
+    mHistMatchingEff->Sumw2();
+    mHistMatchingEff->SetStats(0);
+    mHistMatchingEff->SetNameTitle(name, title);
+    mHistMatchingEff->SetMarkerSize(0.25);
+    mHistMatchingEff->SetLineColor(kBlack);
+    std::string option;
+    if (mHistMatchingEff->InheritsFrom(TH2::Class())) {
+      option = "colz";
+    } else {
+      option = "PE";
+    }
+    histograms().emplace_back(HistInfo{ mHistMatchingEff.get(), option, "" });
+    publish(objectsManager, histograms().back());
+  }
+
+  if (mRebin != 1) {
+    HIST* h1 = dynamic_cast<HIST*>(hMatched->Clone());
+    HIST* h2 = dynamic_cast<HIST*>(hMCH->Clone());
+    h1->Rebin(mRebin);
+    h2->Rebin(mRebin);
+    mHistMatchingEff->Divide(h1, h2);
+    delete h1;
+    delete h2;
+  } else {
+    mHistMatchingEff->Divide(hMatched, hMCH);
+  }
+  mHistMatchingEff->SetNameTitle(name, title);
+}
+
+//_________________________________________________________________________________________
+
+void TracksPostProcessing::createTrackHistos()
+{
+  for (auto& dataSource : mConfig->dataSources) {
+    auto pathMatched = dataSource.plotsPath;
+    auto pathRef = dataSource.refsPath;
+    auto path = dataSource.outputPath;
+    mMatchingEfficiencyPlotters.emplace_back(std::make_unique<MatchingEfficiencyPlotter<TH1>>(dataSource.plotsPath, dataSource.refsPath, dataSource.outputPath, dataSource.name, dataSource.rebin));
+  }
+
+  for (auto& plot : mMatchingEfficiencyPlotters) {
+    plot->publish(getObjectsManager());
+  }
+}
+
+//_________________________________________________________________________________________
+
+void TracksPostProcessing::configure(const boost::property_tree::ptree& config)
+{
+  mConfig = std::make_unique<TracksPostProcessingConfig>(getID(), config);
+
+  createTrackHistos();
+}
+
+//_________________________________________________________________________________________
+
+void TracksPostProcessing::initialize(Trigger, framework::ServiceRegistryRef)
+{
+}
```

Comment:
If you do not clear the plots here, I think the plots from the previous runs will be published at the 2nd START. I could be missing something though...

Reply:
Indeed. To be on the safe side, I added code that un-publishes and destroys all objects before they are re-created and re-published.
On the first START there is simply nothing to un-publish or destroy, but the code handles this correctly.